app/classes/query/base.rb

```diff
@@ -124,38 +120,53 @@
 #    [User]
 #    [Location]
 #
-#  In some cases the array may be ultimately parsed in the scope as a duration
-#  or range of values (e.g. a range of dates, ranks, vote values, etc.). In this
-#  case, two values is the maximum length of the array; more will be ignored.
-#  But for arrays of model instances like [Name], any number of instances or
-#  ids can be passed.
+#  In some cases the scope may be configured to parse the array as a range of
```

Comment:
Would it make sense over to time to replace this usage of array style attributes with hash style attributes, e.g., `{ start: :time, end: :time }`.  Seems like that would help avoid having to dig into the scopes to understand how the array is interpreted.

Reply:
Hmmm - maybe, but maybe not for forms. I was balancing specificity here with the boilerplate i would have to write for a "range" form field, and in this case it seemed way easier to just have a text field that accepts two date strings. But i haven't rewritten the faceted form for Query yet. I will keep this in mind in case it does turn out to be helpful.

Btw, I believe the scopes I wrote are all consistent in parsing ranges, though someone could write one differently.